src/txorphanage.cpp

```diff
@@ -246,6 +246,9 @@ bool TxOrphanage::HaveTxToReconsider(NodeId peer)
 
 void TxOrphanage::EraseForBlock(const CBlock& block)
 {
+    // No orphans, nothing to do
+    if (m_orphans.empty()) return;
```

Comment:
This makes sense on its own I think - it's similar to another draft PR I had (https://github.com/bitcoin-dev-tools/benchcoin/pull/167). If you think it's related and adds anything, feel free to cherry-pick it here.

Reply:
> This makes sense on its own I think - it's similar to another draft PR I had ([bitcoin-dev-tools#167](https://github.com/bitcoin-dev-tools/benchcoin/pull/167)). If you think it's related and adds anything, feel free to cherry-pick it here.

That's cool, will check it in the coming days.
First quick glance it seems that if the mempool is empty; we could avoid calling `removeForBlock` entirely. Which will save us from adding another callback into the validation queue, which will leave space for processing another block during ibd.
Yet, that was a fast check. Should check who is using the `MempoolTransactionsRemovedForBlock` event.